src/cp/cp_control_point.c

```diff
@@ -0,0 +1,406 @@
+    control_points[WT_CONN_CONTROL_POINT_ID_THREAD_8].config_name = "thread_8";
+    WT_ERR(__wt_spin_init(
+      session, &(control_points[WT_CONN_CONTROL_POINT_ID_THREAD_8].lock), "Thread 8"));
+    /* Extra initialization required for action "Wait for trigger". */
+    control_points[WT_CONN_CONTROL_POINT_ID_THREAD_8].action_supported =
+      WT_CONTROL_POINT_ACTION_ID_WAIT_FOR_TRIGGER;
+
+    control_points[WT_CONN_CONTROL_POINT_ID_THREAD_9].init =
+      __wt_control_point_pair_init_always_wait_for_trigger;
+    control_points[WT_CONN_CONTROL_POINT_ID_THREAD_9].pred = NULL; /* Always */
+    control_points[WT_CONN_CONTROL_POINT_ID_THREAD_9].config_name = "thread_9";
+    WT_ERR(__wt_spin_init(
+      session, &(control_points[WT_CONN_CONTROL_POINT_ID_THREAD_9].lock), "Thread 9"));
+    /* Extra initialization required for action "Wait for trigger". */
+    control_points[WT_CONN_CONTROL_POINT_ID_THREAD_9].action_supported =
+      WT_CONTROL_POINT_ACTION_ID_WAIT_FOR_TRIGGER;
+
+    /* After all repeats finish with this. */
+    S2C(session)->control_points = control_points;
+
+err:
+    if (ret != 0)
+        __wt_free(session, control_points);
+    return (ret);
+}
+
+/*
+ * __wt_session_control_point_init_all --
+ *     Initialize all per session control points. Note, one part of this function must be edited for
+ *     each per session control point.
+ *
+ * @param session The session.
+ */
+int
+__wt_session_control_point_init_all(WT_SESSION_IMPL *session)
+{
+#if SESSION_CONTROL_POINTS_SIZE == 0
+    WT_UNUSED(session);
+    return (0);
+#else
+    WT_DECL_RET;
+    WT_CONTROL_POINT_REGISTRY *control_points;
+
+    if (SESSION_CONTROL_POINTS_SIZE == 0)
+        return (0);
+    WT_RET(__wt_calloc_def(session, SESSION_CONTROL_POINTS_SIZE, &control_points));
+
+    /*
+     * This part must be edited. Repeat this for every per session control point.
+     */
+#if 0 /* For example */
+    control_points[WT_SESSION_CONTROL_POINT_ID_EXAMPLE2].init =
+        __wt_control_point_pair_init_skip_wait_for_trigger;
+    control_points[WT_SESSION_CONTROL_POINT_ID_EXAMPLE2].pred =
+        __wt_control_point_pred_skip;
+    control_points[WT_SESSION_CONTROL_POINT_ID_EXAMPLE2].config_name = "example2";
+  WT_ERR(__wt_spin_init(session, &(control_points[WT_SESSION_CONTROL_POINT_ID_EXAMPLE2].lock)));
+#endif
+
+    /* After all repeats finish with this. */
+    session->control_points = control_points;
+
+err:
+    if (ret != 0)
+        __wt_free(session, control_points);
+    return (ret);
+#endif
+}
+
+/*
+ * Control point startup functions: Enable at startup.
+ */
+/*
+ * __wt_conn_control_point_enable_all --
+ *     Enable per connection control points that start enabled. Note, one part of this function must
+ *     be edited for each per connection control point that starts enabled.
+ *
+ * @param conn The connection. @param cfg The configuration strings.
+ */
+int
+__wt_conn_control_point_enable_all(WT_SESSION_IMPL *session, const char **cfg)
```

Comment:
Is it worth merging these init and enable functions? They're called like this:
```
    WT_RET(__wt_conn_control_point_init_all(session));
    WT_RET(__wt_conn_control_point_enable_all(session, cfg));
```
It's also not clear that `enable_all` doesn't enable all but actually the ones that start enabled. I can't think of a better name so merging the two would hide that detail.

Reply:
Keeping them separate is consistent with session. Keeping them separate also makes sense since the meaning is different.
I will rename _enable_all to _enable_all_in_start.